catalog/views.py

```diff
@@ -1,43 +1,27 @@
 from django.shortcuts import render, get_object_or_404, get_list_or_404
-from catalog.models import Category, Brand, Unit, Offer
+from catalog.models import Category, Brand, Offer
 
 
 def index(request):
     brands = Brand.visible.all().order_by('name')
-    units = Unit.visible.filter(parent=None)
-    context = {'brands': brands, 'units': units}
+    categories = Category.visible.filter(parents=None).filter(brand=None)
+    context = {'brands': brands, 'categories': categories}
     return render(request, 'index.html', context=context)
 
 
+def get_category(request, category_slug):
+    category = Category.visible.get(slug=category_slug)
```

Comment:
https://stackoverflow.com/questions/6624530/how-to-use-select-related-with-get-object-or-404